ruby/lib/svix/webhook.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Svix
+    class Webhook
+        def initialize(secret)
+            @secret = Base64.decode64(secret)
+        end
+
+        def verify(payload, headers)
+            msgId = headers["svix-id"]
+            msgSignature = headers["svix-signature"]
+            msgTimestamp = headers["svix-timestamp"]
+            if !msgSignature || !msgId || !msgTimestamp
+                raise WebhookVerificationError, "Missing required headers"
+            end
+
+            toSign = "#{msgId}.#{msgTimestamp}.#{payload}"
+            signature = Base64.encode64(OpenSSL::HMAC.digest(OpenSSL::Digest.new('sha256'), @secret, toSign)).strip()
+
+            passedSignatures = msgSignature.split(" ")
+            passedSignatures.each do |versionedSignature|
+                version, expectedSignature = versionedSignature.split(',', 2)
+                if version != "v1"
+                    next
+                end
+                if signature == expectedSignature
```

Comment:
After reviewing the php version I realised I missed it here too! Needs to use a secure comparison function.

Reply:
Borrowed the secure cmp from Rails ActiveSupport